src/main/java/org/jenkinsci/remoting/util/DurationStyle.java

```diff
@@ -0,0 +1,263 @@
+
+    protected final Matcher matcher(String value) {
+        return this.pattern.matcher(value);
+    }
+
+    /**
+     * Parse the given value to a duration.
+     * @param value the value to parse
+     * @return a duration
+     */
+    public Duration parse(String value) {
+        return parse(value, null);
+    }
+
+    /**
+     * Parse the given value to a duration.
+     * @param value the value to parse
+     * @param unit the duration unit to use if the value doesn't specify one ({@code null}
+     * will default to ms)
+     * @return a duration
+     */
+    public abstract Duration parse(String value, ChronoUnit unit);
+
+    /**
+     * Print the specified duration.
+     * @param value the value to print
+     * @return the printed result
+     */
+    public String print(Duration value) {
+        return print(value, null);
+    }
+
+    /**
+     * Print the specified duration using the given unit.
+     * @param value the value to print
+     * @param unit the value to use for printing
+     * @return the printed result
+     */
+    public abstract String print(Duration value, ChronoUnit unit);
+
+    /**
+     * Detect the style then parse the value to return a duration.
+     * @param value the value to parse
+     * @return the parsed duration
+     * @throws IllegalArgumentException if the value is not a known style or cannot be
+     * parsed
+     */
+    public static Duration detectAndParse(String value) {
+        return detectAndParse(value, null);
+    }
+
+    /**
+     * Detect the style then parse the value to return a duration.
+     * @param value the value to parse
+     * @param unit the duration unit to use if the value doesn't specify one ({@code null}
+     * will default to ms)
+     * @return the parsed duration
+     * @throws IllegalArgumentException if the value is not a known style or cannot be
+     * parsed
+     */
+    public static Duration detectAndParse(String value, ChronoUnit unit) {
+        return detect(value).parse(value, unit);
+    }
+
+    /**
+     * Detect the style from the given source value.
+     * @param value the source value
+     * @return the duration style
+     * @throws IllegalArgumentException if the value is not a known style
+     */
+    public static DurationStyle detect(String value) {
+        if (value == null) {
+            throw new IllegalArgumentException("Value must not be null");
+        }
+        for (DurationStyle candidate : values()) {
+            if (candidate.matches(value)) {
+                return candidate;
+            }
+        }
+        throw new IllegalArgumentException("'" + value + "' is not a valid duration");
+    }
+
+    /**
+     * Units that we support.
+     */
+    enum Unit {
+
+        /**
+         * Nanoseconds.
+         */
+        NANOS(ChronoUnit.NANOS, "ns", Duration::toNanos),
+
+        /**
+         * Microseconds.
+         */
+        MICROS(ChronoUnit.MICROS, "us", (duration) -> duration.toNanos() / 1000L),
```

Comment:
Should really be 
```suggestion
        MICROS(ChronoUnit.MICROS, "µs", (duration) -> duration.toNanos() / 1000L),
```
! Oh well.